platform/CriticalSectionLock.h

```diff
@@ -56,22 +56,57 @@ class CriticalSectionLock {
     }
 
     /** Mark the start of a critical section
-     *     
+     *
      */
+    MBED_DEPRECATED_SINCE("mbed-os-5.7",
+            "This function is inconsistent with RAII and is being removed in the future.")
     void lock()
```

Comment:
I think Sam has concerns about deprecating this function.

Reply:
As I remember Sam had concerns about deprecating whole class